dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java

```diff
@@ -2225,6 +2236,7 @@ public String getResourceIds(TaskDefinition taskDefinition) {
         return StringUtils.join(resourceIds, ",");
     }
 
+    @Transactional
```

Comment:
Will this affect the transaction saved by the workflow? Please verify it.

Reply:
hi @HomminLee , I found that it had a Transactional annotation on the outer layer, I'm not sure if there is a problem with nested transactions.

Reply:
If the outer layer has a try catch of this method, it will have problem.  Anyway, we can avoid this, and remove this Transactional annotation is ok. We need to avoid this method being called in isolation.

Reply:
I had create a demo project to verify nested transactions, variables have: normal, error, try-catch in outer method; normal, error in inner method; REQUIRED propagation(defaut), REQUIRES_NEW propagation.
If use REQUIRED propagation, inner method and outer method will use the same transaction. Any exception occurs, both will rollback. It needs to be explained that the transaction will rollback even if outer method catch the exception thrown by the innter method, cause when the transaction try to commit, it will get the exception: "Transaction rolled back because it has been marked as rollback-only".
If use REQUIRES_NEW propagation, inner method and outer method will use the different transaction. Two transactions will not affect each other. If the outer layer has a try catch of inner method and resolve the exception, the outer method still can commit outer transaction.
I think the default propagation(REQUIRED) is more suitable, what do you think？@caishunfeng @ruanwenjun 
@JinyLeeChina

Reply:
Yes, agree with you. Wait for others' opinions.

Reply:
@caishunfeng @JinyLeeChina Do you have any ideas for this pr, I think this pr is acceptable if it can work well and don't introduce a side effect. 
This prevents someone from calling a method without a transaction(This may happen in secondary development).

Reply:
1. I think there is no need to add transactions for a single operation of the database in the method
2. If the outer layer has transaction control, the inner layer plus transaction control will cause the outer layer's transaction to fail

Reply:
@JinyLeeChina 
1. I only added transaction annotations for methods that contain multiple operations
2. If inner layer has exception, do you think it is more correct to rollback outer layer's transaction? If inner layer has no transaction annotation, outer layer will still receive exception and rollback. If outer layer has no transaction annotation, inner layer rollback but outer layer insert/update successful, isn't this the data is inconsistent?